src/expression_builder.rs

```diff
@@ -17,45 +17,42 @@ pub enum Operand<TS: TypeSystem> {
     Expression(Box<ExpressionBuilder<TS>>),
 }
 
-#[derive(Default, Clone)]
+#[derive(Default, Clone, Debug)]
 pub struct ExpressionBuilder<TS: TypeSystem> {
```

Comment:
Since this is no longer a real builder, should we change it back to ExpressionWriter, or maybe just Expression?

Reply:
I was thinking of that but it is still kinda a builder, or a factory... names are stupid its some sort of design pattern because of the nesting expression aspect.
that's why I left the build function as a public method cause otherwise we would be passing the ExecutionContext to all of the... nvm I just realized I got rid of the need for the execution context but never got rid of the argument

yeah the name I was thinking of was `Expression` Ill change it to that

Reply:
Do you also agree with converting it to an enum?